actix/src/sec-7-stream.md

```diff
@@ -1 +1,305 @@
-[WIP] 
\ No newline at end of file
+# Streams
+
+Actix allows the handling of Tokio Streams to act as a source of events
+or messages to actors. This is commonly used with networking servers
+and clients, for communication over TCP, UDP and UnixSockets.
+
+At it's core, Actix when using streams provides a mechanism for taking
+whole message frames from a stream, and converting these to messages which
+are sent to your Actor's handlers.
+
+To show how these parts are created, we'll create a TCP echo server.
+
+## Wrapping a stream of events
+
+A stream provides a series of frames to Actix which reperesent some event
+or input provided by the stream. The first method to handle these inputs
+from a stream, is to asynchronously map over the inputs and turn these into
+messages that can be sent to an Actor.
+
+The best example for this is a TcpListener, which provides a stream
```

Comment:
No, it is not the best. The best would to use corresponding trait [StreamHandler](https://docs.rs/actix/0.7.10/actix/trait.StreamHandler.html)

You need to show that you might need handle errors or/and EOF

Reply:
I disagree here. Most examples seem to be all oriented around the TCP use case, and it will be a common case for users requesting this. I think that extensive documentation of the details and errors/EOF should be in the module-level documentation. IMO the book is "here is how you put the parts together for useful things" and the module-level is "here is every single implementation detail and idea you need to understand". So how about we improve the StreamHandler examples and documentation in the module in parallel? 

PS: I also can't find any non-trivial examples of StreamHandlers that are not TCP based anyway, so that's a barrier to me working this out ....

Reply:
What I mean is that the `best` is subjective word and there is no place for opinion in docs.
`TcpListener` is just

In the context of current paragraph, it is not clear whether you mean `best` as if example of Stream
Or you mean it as example of handling of stream

Reply:
>I also can't find any non-trivial examples of StreamHandlers that are not TCP based anyway, so that's a barrier to me working this out .

Btw that's because TCP is the most common protocol that can be used with framed messages and therefore it is easy to represent TCP stream as `futures::Stream`

Anyway we should talk about `StreamHandler` since it is more effective comparing to your current approach